e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -436,6 +418,26 @@ $keyVaultName = az deployment group show -g $ResourceGroup -n $deploymentName --
 $instrumentationKey = az deployment group show -g $ResourceGroup -n $deploymentName --query 'properties.outputs.instrumentationKey.value' --output tsv
 $iotHubName = az deployment group show -g $ResourceGroup -n $deploymentName --query 'properties.outputs.hubName.value' --output tsv
 
+#################################################################################################################################################
+# Configure an AAD app and create self signed certs and get the bytes to generate more content info.
+#################################################################################################################################################
+Write-Host "`nCreating App Registration $logAnalyticsAppRegnName"
+$logAnalyticsAppRegUrl = "http://$logAnalyticsAppRegnName"
+az ad sp create-for-rbac -n $logAnalyticsAppRegUrl --role "Reader" --scope $resourceGroupId --output none
+$logAnalyticsAppId = az ad app list --display-name $logAnalyticsAppRegnName --query "[?displayName=='$logAnalyticsAppRegnName'].appId" --output tsv
+Write-Host "`nApplication $logAnalyticsAppRegnName with Id $logAnalyticsAppId was created successfully."
+
+#################################################################################################################################################
+# Configure an AAD app to perform IoT hub data actions.
```

Comment:
Newly added for aad auth tests. az ad sp create-for-rbac ensures we don't need any retry logic.